### PR TITLE
`update-repo-wrapper`

### LINKED ISSUE #435: Document SOP for updating the repo `sh run.sh make update_repo`
New plan: [`odkrunner`](https://github.com/gouttegd/odkrunner)

Action items:
- [ ] #719
- [x] @twhetzel to sign off on this 
- [x] @joeflack4 to sign off on this
- [ ] @joeflack4 to add this to the documentation

<details><summary>Original plan (obsolete)</summary>
<p>

## Until 20th February

1. Run:
    ```
    sh run.sh make update_repo
    ```

This is using `run.sh.conf` which has all the relevant environment variables, including `ODK_TAG=dev`. Contrary to what I told you @joeflack4 you cannot just pass `ODK_TAG=dev` to run.sh, because run.sh.conf has precedence.

## After 20th February

1. Edit `src/ontology/run.sh.conf` and set `ODK_TAG=latest`
1. Run:
    ```
    sh run.sh make update_repo
    ```
1. Edit `src/ontology/run.sh.conf` and set `ODK_TAG=dev`
1. Commit changes.

</p>
</details>

### PR DESCRIPTION
**Pull request by joeflack4**
resolves #435

## Overview
Convenience command to automate some otherwise manual steps when running `update_repo`. Expected behavior is that it (i) temporarily updates the `ODK_TAG` used, and (ii) runs `update_repo` twice, until a fix for https://github.com/INCATools/ontology-development-kit/issues/989 is in ODK.

## Pre-merge checklist
- [ ] Docs
   - `docs/` have been added/updated **OR**
   - No updates to the docs necessary after careful consideration.
- [x] QC
   - `sh run.sh make build-mondo-ingest` has been run on this branch (after `docker pull obolibrary/odkfull:dev), and no errors occurred **OR**
   -  No functional (code-related) changes to the pipeline are suggested, so no re-run is necessary.
- [x] Account for any new packages
  - Open an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) to request addition. Can also open a PR. Python package changes should occur in [requirements.txt.full](https://github.com/INCATools/ontology-development-kit/blob/master/requirements.txt.full), and non-Python packages in [Dockerfile](https://github.com/INCATools/ontology-development-kit/blob/master/Dockerfile).
- [ ] Reviewed
   - Has been sufficiently reviewed by at least one review from a different team member of the Mondo Technical team.

## Changes
- Update: mondo-ingest.Makefile: Added goal 'update-repo-wrapper' to automate some steps required to run 'update_repo'.

### CODE REVIEW

File: src/ontology/mondo-ingest.Makefile [hunk 1]
```diff
@@ -575,6 +575,20 @@ update-jinja-sparql-queries:
 	python3 $(SCRIPTSDIR)/ordo_mapping_annotations/create_sparql__ordo_replace_annotation_based_mappings.py
 	python3 $(SCRIPTSDIR)/ordo_mapping_annotations/create_sparql__ordo_mapping_annotations_violation.py
 
+# update-repo-wrapper
+UPDATE_REPO_ODK_TAG := latest
```
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:579]: ## `UPDATE_REPO_ODK_TAG`
Set to `latest` unless the user passes `ODK_TAG`, in which value is set to that.

`ODK_TAG` override usage:
- Works: `sh run.sh make ODK_TAG=v1.4.3 update-repo-wrapper`
- Fails: `ODK_TAG=v1.4.3 sh run.sh make update-repo-wrapper`. This doesn't cause the goal to fail, but the environmental variable doesn't get read.

File: src/ontology/mondo-ingest.Makefile [hunk 2]
```diff
@@ -575,6 +575,20 @@ update-jinja-sparql-queries:
...
+ifdef ODK_TAG
+	UPDATE_REPO_ODK_TAG := $(ODK_TAG)
+endif
+
+.PHONY: update-repo-wrapper
+update-repo-wrapper:
```
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:585]: ## `update-repo-wrapper`
1. Temporarily updates `run.sh.conf` `ODK_TAG` to value of `UPDATE_REPO_ODK_TAG`
2. Runs `update_repo` twice
- We can change this if ever https://github.com/INCATools/ontology-development-kit/issues/989 gets fixed.

**Review by twhetzel — COMMENTED** [review]
I have to defer this review and whether this is needed (at all and/or for the January release) to @matentzn. 

Does this PR close all tasks in https://github.com/monarch-initiative/mondo-ingest/issues/435?

**Comment by matentzn** [comment]
There has been a recent development that makes this PR obsolete: The ODK runner: https://github.com/gouttegd/odkrunner

I am using it only now for interacting with ODK, and it will get rid of any issues, especially running the update multiple times etc.. Just install it and from then on:

odk make xyz instead of sh run.sh make xyz, and it will still recognise run.sh.conf!

**Pull request closed without merging**